src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1252,6 +1262,39 @@ private static DkmClrCompilationResultFlags GetLocalResultFlags(LocalSymbol loca
                 isIteratorOrAsyncMethod = GeneratedNames.GetKind(containingType.Name) == GeneratedNameKind.StateMachineType;
             }
 
+            var parameterNames = PooledHashSet<string>.GetInstance();
+            var parameterNamesInOrder = ArrayBuilder<string>.GetInstance();
+            // For version before .NET 4.5, we cannot find the sourceMethod properly:
+            // The source method coincides with the original method in the case.
+            // Here, we have to get parameters from the containingType.
+            // This does not guarantees the proper order of parameters.
+            if (isIteratorOrAsyncMethod && method == sourceMethod)
+            {
+                Debug.Assert(IsDisplayClassType(containingType));
+                foreach (var field in containingType.GetMembers().OfType<FieldSymbol>())
```

Comment:
This line existed before this PR but typically member type checks are handled by checking `Kind` to allocating enumerators:
```
foreach (var member in containingType.GetMembers())
{
    if (member.Kind != SymbolKind.Field)
    {
        continue;
    }
    var field = (FieldSymbol)member;
    ...
}
```

Reply:
OK. Thanks!

---
In reply to: [135917729](https://github.com/dotnet/roslyn/pull/21707#discussion_r135917729) [](ancestors = 135917729)